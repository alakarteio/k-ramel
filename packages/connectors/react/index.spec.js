/* eslint-env jest */
import * as lib from './dist/index.es'
import tests from './misc/test-suite'

tests(lib)
