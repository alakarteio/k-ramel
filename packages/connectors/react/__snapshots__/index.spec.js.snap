// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react should add listeners at mount 1`] = `
Object {
  "dispatch": Array [
    Array [
      "@@krml/LISTENERS>ADDED",
    ],
    Array [
      "@@krml/LISTENERS>REMOVING",
    ],
  ],
}
`;

exports[`react should dispatch events at mount with a given name 1`] = `
Object {
  "dispatch": Array [
    Array [
      "@@krml/LISTENERS>ADDED>my-name",
    ],
    Array [
      "@@krml/LISTENERS>REMOVING>my-name",
    ],
  ],
}
`;

exports[`react should inject store to a component 1`] = `"<div>a label</div>"`;

exports[`react should not throw an exception if no provider is part of parents 1`] = `
Array [
  Array [
    "[k-ramel/react] Error in %cinject%c for the component %cComponent%c
	> The store needs to be provided by an ancestor of this component.
	> You can use %cprovider%c from %c@k-ramel/react%c or %cProvider%c from %creact-redux%c.

Check the documentation for an example at https://github.com/unirakun/k-ramel/tree/master/packages/connectors/react#examples--api
",
    "font-weight: bolder; font-style: italic;",
    "",
    "font-weight: bolder; font-style: italic;",
    "",
    "font-weight: bolder; font-style: italic;",
    "",
    "font-weight: bolder; font-style: italic;",
    "",
    "font-weight: bolder; font-style: italic;",
    "",
    "font-weight: bolder; font-style: italic;",
    "",
  ],
]
`;

exports[`react should provide store as context 1`] = `
Object {
  "html": "<div>{\\"this\\":\\"is my store\\"}</div>",
}
`;

exports[`react should refresh when props changes 1`] = `"<div>a label-2</div>"`;

exports[`react should refresh when store changes 1`] = `"<div>an other label</div>"`;
